src/core.rs

```diff
@@ -2124,3 +2200,21 @@ fn swap_remove_if<T, F>(vec: &mut Vec<T>, pred: F) -> Option<T>
         None
     }
 }
+
+#[cfg(not(feature = "use-mock-crust"))]
+fn restart_crust_service(service: &mut Service, sender: crust::CrustEventSender) {
+    use std::mem;
+
+    let _ = mem::replace(service,
```

Comment:
Yes, but what I'd like the best is if even the real `crust::Service` has a `restart` method. Then we wouldn't need this hack at all.